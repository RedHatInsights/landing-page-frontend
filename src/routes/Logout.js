import React from 'react';
import { Title, Button } from '@patternfly/react-core';

import './Logout.scss';

const onButtonClick = (url) => {
    window.location.href = `./${url}`;
};

const Logout = () => {
    return (
        <section className="pf-l-page__main-section pf-c-page__main-section ins-c-page__logout">
            <Title headingLevel='h1' size='3xl'> You have logged out. </Title>
            <Button variant='link' onClick={ () => onButtonClick('') }>Take me home</Button>
        </section>
    );
};

export default Logout;
