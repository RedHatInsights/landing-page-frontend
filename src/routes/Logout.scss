@import '~@red-hat-insights/insights-frontend-components/Utilities/_mixins';

.ins-c-page__logout {
    justify-content: center;
    flex-direction: column;
    text-align: center;

    h1 { @include rem('margin-bottom', 30px); }
    button { @include rem('font-size', 20px); }
}