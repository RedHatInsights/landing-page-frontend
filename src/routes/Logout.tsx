import { Button } from '@patternfly/react-core/dist/dynamic/components/Button';
import { Title } from '@patternfly/react-core/dist/dynamic/components/Title';
import React from 'react';

import './Logout.scss';

const onButtonClick = (url: string) => {
  window.location.href = `./${url}`;
};

const Logout = () => (
  <section className="pf-v5-l-page__main-section pf-v5-c-page__main-section land-c-page__logout">
    <Title headingLevel="h1" size="3xl">
      You have logged out.
    </Title>
    <Button variant="link" onClick={() => onButtonClick('')}>
      Take me home
    </Button>
  </section>
);

export default Logout;
