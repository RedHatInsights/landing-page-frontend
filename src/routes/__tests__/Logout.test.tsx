import React from 'react';
import Logout from '../Logout';
import { render, screen } from '@testing-library/react';

describe('Logout component', () => {
  it('should render correctly', () => {
    const { container } = render(<Logout />);
    expect(container).toMatchSnapshot();
  });

  it('click: take me home', () => {
    render(<Logout />);
    screen.getByText('Take me home').click();
    expect(window.location.pathname).toBe('/');
  });
});
