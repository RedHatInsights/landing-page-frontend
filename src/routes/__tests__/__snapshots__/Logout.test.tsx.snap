// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout component should render correctly 1`] = `
<div>
  <section
    class="pf-v5-l-page__main-section pf-v5-c-page__main-section land-c-page__logout"
  >
    <h1
      class="pf-v5-c-title pf-m-3xl"
      data-ouia-component-id="OUIA-Generated-Title-1"
      data-ouia-component-type="PF5/Title"
      data-ouia-safe="true"
    >
      You have logged out.
    </h1>
    <button
      aria-disabled="false"
      class="pf-v5-c-button pf-m-link"
      data-ouia-component-id="OUIA-Generated-Button-link-1"
      data-ouia-component-type="PF5/Button"
      data-ouia-safe="true"
      type="button"
    >
      Take me home
    </button>
  </section>
</div>
`;
